framework/include/fvics/FVInitialConditionBase.h

```diff
@@ -0,0 +1,83 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseObject.h"
+#include "FunctionInterface.h"
+#include "UserObjectInterface.h"
+#include "PostprocessorInterface.h"
+#include "Restartable.h"
+#include "BlockRestrictable.h"
+#include "DependencyResolverInterface.h"
+#include "MooseTypes.h"
+#include "ADFunctorInterface.h"
+
+class SystemBase;
+class MooseVariableFieldBase;
+namespace libMesh
+{
+class Point;
+}
+
+/**
+ * description
+ */
+class FVInitialConditionBase : public MooseObject,
+                               public BlockRestrictable,
+                               public FunctionInterface,
+                               public Restartable,
+                               public DependencyResolverInterface,
+                               public ADFunctorInterface
+{
+public:
+  /**
+   * Constructor
+   *
+   * @param parameters The parameters object holding data for the class to use.
+   */
+  FVInitialConditionBase(const InputParameters & parameters);
+
+  virtual ~FVInitialConditionBase();
+
+  static InputParameters validParams();
+
+  /**
+   * retrieves the MOOSE variable that this initial condition acts upon
+   */
+  virtual MooseVariableFieldBase & variable() = 0;
+
+  /**
+   * Workhorse method for projecting the initial conditions for boundary restricted initial
```

Comment:
Only boundary restricted?

Reply:
I wanted to use block :(, no boundary restriction for us at this point